### PR TITLE
feat(api): new CommonAudit api to support dual deployment model

### PR DESCRIPTION
**Pull request by hbradfield**
https://github.ibm.com/IBMPrivateCloud/roadmap/issues/38738

**Prereq**s:
- `ibm-common-services` namespace
- `ibm-cert-manager-operator` installed on your cluster
- Secret for artifactory scratch repo named `docker-scratch`: `oc create secret -n ibm-common-services docker-registry docker-scratch  --docker-server=hyc-cloud-private-scratch-docker-local.artifactory.swg-devops.com --docker-username=<REPLACE_ME>  --docker-password=<<REPLACE_ME> --docker-email=<REPLACE_ME>`

**Testing**: 
- Replace operator container image in 3.7.0 csv with `hyc-cloud-private-scratch-docker-local.artifactory.swg-devops.com/audit-operator:test`
- Update AuditLogging CR and CommonAudit CR to use `fluentd.imageRegistry: hyc-cloud-private-scratch-docker-local.artifactory.swg-devops.com/ibmcom` 
- Configure rest of AuditLogging CR and CommonAudit CR 
- `make install`

Note: If fluentd cannot pull scratch images, add `docker-scratch` as an `ImagePullSecret` to the deploy/dameonset

### CODE REVIEW

**Comment by lauriet-ibm** [comment]
@hbradfield What's this field for https://github.com/IBM/ibm-auditlogging-operator/blob/1624751d055d55fb5ec19ce74875b41f4c71c87e/pkg/apis/operator/v1/commonaudit_types.go#L30

**Comment by hbradfield** [comment]
`Size` is a placeholder for the new bedrock sizing. I can take it out of this PR since it seems like they still haven't fully arched how we want to approach sizing

File: pkg/apis/operator/v1/commonaudit_types.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	ClusterIssuer                string                 `json:"clusterIssuer,omitempty"`
+	Fluentd                      CommonAuditSpecFluentd `json:"fluentd,omitempty"`
+	Outputs                      CommonAuditSpecOutputs `json:"outputs,omitempty"`
+}
+
+// CommonAuditSpecFluentd defines the desired state of Fluentd
+type CommonAuditSpecFluentd struct {
+	ImageRegistry string                   `json:"imageRegistry,omitempty"`
+	PullPolicy    string                   `json:"pullPolicy,omitempty"`
+	Replicas      int                      `json:"replicas,omitempty"`
```
Comment (hbradfield) [inline pkg/apis/operator/v1/commonaudit_types.go:39]: @lauriet-ibm im not sure where we are supposed to put the "replicas" field...according to this doc it should be in the outermost spec https://github.ibm.com/IBMPrivateCloud/BedrockServices/blob/master/OverallRequiredCapabilities/OperatorChangesForSizing.md#crd-spec-single-workload

**Comment by hbradfield** [comment]
@lauriet-ibm pushed the latest operator image to scratch and updated steps to test above

File: deploy/olm-catalog/ibm-auditlogging-operator/3.7.0/ibm-auditlogging-operator.v3.7.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -373,19 +472,17 @@ spec:
                 - ibm-auditlogging-operator
                 env:
                 - name: WATCH_NAMESPACE
-                  valueFrom:
-                    fieldRef:
-                      fieldPath: metadata.annotations['olm.targetNamespaces']
```
Comment (hbradfield) [inline deploy/olm-catalog/ibm-auditlogging-operator/3.7.0/ibm-auditlogging-operator.v3.7.0.clusterserviceversion.yaml:378]: I'll need to revert back to `metadata.annotations['olm.targetNamespaces']` once I test with odlm deploying the operator as `installMode: cluster`

Comment (hbradfield) [reply deploy/olm-catalog/ibm-auditlogging-operator/3.7.0/ibm-auditlogging-operator.v3.7.0.clusterserviceversion.yaml:378]: Installing to `openshift-operators` may break us in an upgrade scenario

**Comment by hbradfield** [comment]
need to fix a test

**Comment by hbradfield** [comment]
@lauriet-ibm i went through the files and fixed a reference to a scratch image. if you could just take one more pass through the files and see if i missed anything id appreciate it. we can merge if it's all good

**Pull request merged**